tests/pyapi/test_dependencies.py

```diff
@@ -38,3 +40,20 @@ def test_dep_failed(hq_env: HqEnv):
     assert table.get_row_value("1") == "CANCELED"
     assert table.get_row_value("2") == "CANCELED"
     assert table.get_row_value("3") == "FINISHED"
+
+
+def test_kill_worker_with_deps(hq_env: HqEnv):
```

Comment:
This test succeeds on `main`.

Reply:
I know, I am not not able to trigger it on main. But as I have written this test I just put it in repo

Reply:
Ok :)